x-pack/plugins/endpoint/public/applications/endpoint/index.tsx

```diff
@@ -8,28 +8,37 @@ import * as React from 'react';
 import ReactDOM from 'react-dom';
 import { CoreStart, AppMountParameters } from 'kibana/public';
 import { I18nProvider, FormattedMessage } from '@kbn/i18n/react';
-import { Route, BrowserRouter, Switch } from 'react-router-dom';
-import { Provider } from 'react-redux';
+import { Route, Switch, BrowserRouter, useLocation } from 'react-router-dom';
+import { Provider, useDispatch } from 'react-redux';
 import { Store } from 'redux';
+import { memo } from 'react';
 import { appStoreFactory } from './store';
 import { AlertIndex } from './view/alerts';
+import { AppAction } from './store/action';
+import { EndpointAppLocation } from './types';
 
 /**
  * This module will be loaded asynchronously to reduce the bundle size of your plugin's main bundle.
  */
 export function renderApp(coreStart: CoreStart, { appBasePath, element }: AppMountParameters) {
   coreStart.http.get('/api/endpoint/hello-world');
-
   const [store, stopSagas] = appStoreFactory(coreStart);
 
-  ReactDOM.render(<AppRoot basename={appBasePath} store={store} />, element);
+  ReactDOM.render(<AppRoot store={store} basename={appBasePath} />, element);
 
   return () => {
     ReactDOM.unmountComponentAtNode(element);
     stopSagas();
   };
 }
 
+const RouteCapture = memo(({ children }) => {
+  const location: EndpointAppLocation = useLocation();
+  const dispatch: (action: AppAction) => unknown = useDispatch();
+  dispatch({ type: 'userChangedUrl', payload: location });
```

Comment:
I had mentioned this in my prior comment and just wanted to again make everyone aware - this will dispatch the action potentially BEFORE the component for the matched route is rendered. I think that is ok, but just wanted everyone to be aware of that behaviour.